lib/src/observe/coap_observe_relation.dart

```diff
@@ -14,15 +14,16 @@ class CoapObserveRelation {
   /// The observing endpoint
```

Comment:
I think this file could probably be adjusted a bit:

```dart
/*
 * Package : Coap
 * Author : S. Hamblett <steve.hamblett@linux.com>
 * Date   : 31/10/2017
 * Copyright :  S.Hamblett
 */

part of coap;

/// Represents a relation between a client endpoint and a resource on this
/// server.
class CoapObserveRelation {
  /// Constructs a new observe relation.
  ///
  /// Takes the observing [endpoint], the observed [resource], and the
  /// [exchange] that tries to establish the observe relation as arguments.
  CoapObserveRelation(this.cfg, CoapObservingEndpoint endpoint,
      CoapIResource resource, CoapExchange exchange)
      : _endpoint = endpoint,
        _resource = resource,
        _exchange = exchange,
        established = true;

  DefaultCoapConfig cfg;

  final CoapObservingEndpoint _endpoint;

  /// Source endpoint of the observing endpoint
  InternetAddress? get source => _endpoint.endpoint;
  final CoapIResource _resource;

  /// The resource
  CoapIResource? get resource => _resource;
  final CoapExchange _exchange;

  /// The exchange
  CoapExchange get exchange => _exchange;

  /// Current control notification
  CoapResponse? currentControlNotification;

  /// Next control notification
  CoapResponse? nextControlNotification;

  /// Key
  String get key => '$source#${_exchange.request?.tokenString}';

  /// A value indicating if this relation has been established
  bool established;
  DateTime _interestCheckTime = DateTime.now();
  int _interestCheckCounter = 1;

  /// The notifications that have been sent, so they can be
  /// removed from the Matcher.
  final Queue<CoapResponse?> _notifications = Queue<CoapResponse?>();

  /// Cancel this observe relation.
  void cancel() {
    // Stop ongoing retransmissions
    if (_exchange.response != null) {
      _exchange.response!.isCancelled = true;
    }
    established = false;
    _resource.removeObserveRelation(this);
    _endpoint.removeObserveRelation(this);
    _exchange.complete = true;
  }

  /// Cancel all observer relations that this server has
  /// established with this's realtion's endpoint.
  void cancelAll() {
    _endpoint.cancelAll();
  }

  /// Notifies the observing endpoint that the resource has been changed.
  void notifyObservers() {
    // Makes the resource process the same request again
    _resource.handleRequest(_exchange);
  }

  /// Check
  bool check() {
    var check = false;
    final now = DateTime.now();
    check = check ||
        _interestCheckTime
            .add(Duration(milliseconds: cfg.notificationCheckIntervalTime))
            .isBefore(now);
    check = check ||
        (++_interestCheckCounter >= cfg.notificationCheckIntervalCount);
    if (check) {
      _interestCheckTime = now;
      _interestCheckCounter = 0;
    }
    return check;
  }

  /// Add a notification
  void addNotification(CoapResponse notification) {
    _notifications.add(notification);
  }

  /// Clear notifications
  Iterable<CoapResponse?> clearNotifications() {
    Iterable<CoapResponse?> list;
    list = _notifications.toList();
    _notifications.clear();
    return list;
  }
}
```

Reply:
Fixed: https://github.com/shamblett/coap/pull/76/commits/806d2aabc363905fe1dde8efcbf881d42f3118a3